staging/src/k8s.io/client-go/util/workqueue/parallelizer.go

```diff
@@ -50,3 +50,36 @@ func Parallelize(workers, pieces int, doWorkPiece DoWorkPieceFunc) {
 	}
 	wg.Wait()
 }
+
+// ConditionFunc returns true if the condition is satisfied, or an error
+// if the loop should be aborted.
+type ConditionFunc func() (done bool, err error)
+
+// ParallelizeUntil is a very simple framework that allow for parallelizing
+// N independent pieces of work until the condition function return true.
+func ParallelizeUntil(workers, pieces int, doWorkPiece DoWorkPieceFunc, condition ConditionFunc) {
+	toProcess := make(chan int, pieces)
+	for i := 0; i < pieces; i++ {
+		toProcess <- i
+	}
+	close(toProcess)
+	if pieces < workers {
+		workers = pieces
+	}
+	wg := sync.WaitGroup{}
+	wg.Add(workers)
+	for i := 0; i < workers; i++ {
+		go func() {
+			defer utilruntime.HandleCrash()
+			defer wg.Done()
+			for piece := range toProcess {
+				// Return and abort if the condition is satisfied.
+				if done, err := condition(); done && err == nil {
```

Comment:
btw, as an util, do we accept nil conditionFunc ?

Reply:
Yes,we should handle it. I'll follow misterikkit's comment and update it later.